cmd/metal-api/internal/service/ip-service.go

```diff
@@ -150,76 +149,74 @@ func (r *ipResource) findIP(request *restful.Request, response *restful.Response
 	id := request.PathParameter("id")
 
 	ip, err := r.ds.FindIPByID(id)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
-		return
-	}
-	err = response.WriteHeaderAndEntity(http.StatusOK, v1.NewIPResponse(ip))
 	if err != nil {
-		r.log.Errorw("failed to send response", "error", err)
+		r.sendError(request, response, defaultError(err))
 		return
 	}
+
+	r.send(request, response, http.StatusOK, v1.NewIPResponse(ip))
 }
 
 func (r *ipResource) listIPs(request *restful.Request, response *restful.Response) {
 	ips, err := r.ds.ListIPs()
-	if checkError(request, response, utils.CurrentFuncName(), err) {
+	if err != nil {
+		r.sendError(request, response, defaultError(err))
 		return
 	}
 
 	result := []*v1.IPResponse{}
 	for i := range ips {
 		result = append(result, v1.NewIPResponse(&ips[i]))
 	}
-	err = response.WriteHeaderAndEntity(http.StatusOK, result)
-	if err != nil {
-		r.log.Errorw("failed to send response", "error", err)
-		return
-	}
+
+	r.send(request, response, http.StatusOK, result)
 }
 
 func (r *ipResource) findIPs(request *restful.Request, response *restful.Response) {
 	var requestPayload datastore.IPSearchQuery
 	err := request.ReadEntity(&requestPayload)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
+	if err != nil {
+		r.sendError(request, response, httperrors.BadRequest(err))
 		return
 	}
 
 	var ips metal.IPs
 	err = r.ds.SearchIPs(&requestPayload, &ips)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
+	if err != nil {
+		r.sendError(request, response, defaultError(err))
 		return
 	}
 
 	result := []*v1.IPResponse{}
 	for i := range ips {
 		result = append(result, v1.NewIPResponse(&ips[i]))
 	}
-	err = response.WriteHeaderAndEntity(http.StatusOK, result)
-	if err != nil {
-		r.log.Errorw("failed to send response", "error", err)
-		return
-	}
+
+	r.send(request, response, http.StatusOK, result)
 }
 
 func (r *ipResource) freeIP(request *restful.Request, response *restful.Response) {
 	id := request.PathParameter("id")
 
 	ip, err := r.ds.FindIPByID(id)
-	if checkError(request, response, utils.CurrentFuncName(), err) {
-		return
-	}
-	if checkError(request, response, utils.CurrentFuncName(), validateIPDelete(ip)) {
+	if err != nil {
+		r.sendError(request, response, defaultError(err))
 		return
 	}
-	if checkError(request, response, utils.CurrentFuncName(), r.actor.releaseIP(*ip)) {
+
+	err = validateIPDelete(ip)
+	if err != nil {
+		r.sendError(request, response, httperrors.BadRequest(err))
 		return
 	}
 
-	err = response.WriteHeaderAndEntity(http.StatusOK, v1.NewIPResponse(ip))
+	err = r.actor.releaseIP(*ip)
```

Comment:
both function calls have not been here before

Reply:
Was there but hidden inside the checkError func call. :(